### PR TITLE
Allow decimals in the local alert radius.

### PR DESCRIPTION
**Pull request by dracos**
Fixes https://github.com/mysociety/societyworks/issues/3278

### CODE REVIEW

File: perllib/FixMyStreet/App/Controller/Alert.pm [hunk 1]
```diff
@@ -101,7 +101,11 @@ sub rss : Private {
     elsif ( $feed =~ /^local:([\d\.-]+):([\d\.-]+)$/ ) {
         my $distance = $c->forward('get_distance');
         $url = $c->cobrand->base_url() . '/rss/l/' . $1 . ',' . $2;
-        $url .= "/$distance" if $distance;
+        if ($distance) {
+            $distance = mySociety::Locale::in_gb_locale { sprintf("%f", $distance); };
```
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/App/Controller/Alert.pm:105]: Ah, this is used in various places in the codebase - in a non-English locale, like France, an internal number will be stringified as "5,2". But we need it to have a decimal point.

File: perllib/FixMyStreet/App/Controller/Alert.pm [hunk 2]
```diff
@@ -544,10 +548,10 @@ sub setup_request : Private {
 sub get_distance : Private {
     my ($self, $c) = @_;
 
-    my $distance = $c->get_param('distance') || '';
-    $distance = 0 unless $distance =~ /^\d+$/;
+    my $distance = $c->get_param('distance') || 0;
+    $distance = 0 unless $distance =~ /^\d*([,.]\d+)?$/;
     $distance = 150 if $distance > 150; # Match Gaze maximum
-    return $distance;
+    return $distance + 0;
```
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/App/Controller/Alert.pm:554]: To make sure it is internally a number, not a string of a number.

File: perllib/FixMyStreet/App/Controller/Rss.pm [hunk 3]
```diff
@@ -111,7 +111,7 @@ sub local_problems_pc_distance : Path('pc') : Args(2) {
 
 }
 
-sub local_problems_dist : LocalRegex('^(n|l)/([\d.-]+)[,/]([\d.-]+)/(\d+)$') {
+sub local_problems_dist : LocalRegex('^(n|l)/([\d.-]+)[,/]([\d.-]+)/(\d+(?:\.\d+)?)$') {
```
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/App/Controller/Rss.pm:114]: Yes, get_distance allows people to enter decimals as commas, but then it's always a decimal point by the time it reaches the URL (and then here). 
RSS feeds of that form come here directly, it's a controller, so not called from anywhere. The Alert page will construct an RSS in this form if you supplied a distance there.

**Review by nephila-nacrea — APPROVED** [review]

File: perllib/FixMyStreet/App/Controller/Alert.pm (hunk 2 above)
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/App/Controller/Alert.pm:554]: Yes, it does do that, but in a non-English locale a decimal number could eg stringify with a comma, not a point. I know we call `in_gb_locale` above, but if the input is already converted to a string before that point (with a comma), that call won't change anything, it's already a string. So we need to make sure it's a number before it gets there.
I'm not sure it /actually/ matters here, there's no `use locale` but I think that can have spooky action at a distance so better to be safe.

**Pull request merged**